### ISSUE TITLE
Failure with Query targets in project directories...

### ISSUE THREAD

**Issue by fujigon** [bazelbuild/intellij#1540]
### Problem I faced

In Windows, "derive_targets_from_directories" feature leads to fail sync.

```
Syncing project: Sync (incremental)...
Updating VCS...
Running Bazel info...
Command: C:\Tools\bazel-2.0.0-windows-x86_64\bazel.exe info --tool_tag=ijwb:IDEA:community --curses=no --color=yes --progress_in_terminal_title=no --

Command: git diff --name-status --no-renames e444771dbfa105e7649132da7b921f7c26ec5fd8

Computing VCS working set...
  _BUILD.bazel (modified)
  abc\abc121\a\BUILD.bazel (modified)
  abc\abc121\b\BUILD.bazel (modified)
  abc\abc121\c\BUILD.bazel (modified)
  abc\abc122\a\BUILD.bazel (modified)
  abc\abc122\b\BUILD.bazel (modified)
  abc\abc122\c\BUILD.bazel (modified)
  abc\abc122\d\BUILD.bazel (modified)
  abc\abc123\a\BUILD.bazel (modified)
  abc\abc123\b\BUILD.bazel (modified)
  abc\abc123\c\BUILD.bazel (modified)
  abc\abc123\d\BUILD.bazel (modified)
  abc\abc124\a\BUILD.bazel (modified)
  abc\abc124\b\BUILD.bazel (modified)
  abc\abc124\c\BUILD.bazel (modified)
  abc\abc124\d\BUILD.bazel (modified)
  abc\abc125\a\BUILD.bazel (modified)
  abc\abc125\b\BUILD.bazel (modified)
  abc\abc125\c\BUILD.bazel (modified)
  abc\abc125\d\BUILD.bazel (modified)
  (and 139 more)

Querying targets in project directories...
Command: C:\Tools\bazel-2.0.0-windows-x86_64\bazel.exe query --tool_tag=ijwb:IDEA:community --output=label_kind --keep_going "attr(\"tags\", \"^((?!manual).)*$\", //...:all - //bazel-bin/...:all - //bazel-genfiles/...:all - //bazel-out/...:all - //bazel-testlogs/...:all - //bazel-atcoder/...:all - //.ijwb/...:all)" --

Sync finished
Sync failed
```

#### Environment
OS : WIndows 10
IntelliJ: 2019.2.4
bazel: 2.0.0
bazel plugin: current master

### Investigation
With print-debug, found that query command [BlazeQueryDirectoryToTargetProvider](https://github.com/bazelbuild/intellij/blob/e2229390e644cbc90f7fa51f56eb6fe8d22895a6/base/src/com/google/idea/blaze/base/dependencies/BlazeQueryDirectoryToTargetProvider.java#L65) fails with error message 

```
ERROR: Error while parsing 'attr(tags, ^((?!manual).)*$, //...:all - //bazel-bin/...:all - //bazel-genfiles/...:all - //bazel-out/...:all - //bazel-testlogs/...:all - //bazel-atcoder/...:all - //.ijwb/...:all)': syntax error at '^ ( ('
```

I also have MacOS environment, and tested the same settings, and it works.
However, when the "just copied" the command from the sync output and executed
```
/usr/local/bin/bazel query --tool_tag=ijwb:IDEA:community --output=label_kind --keep_going "attr(\"tags\", \"^((?!manual).)*$\", //...:all - //bazel-bin/...:all - //bazel-genfiles/...:all - //bazel-out/...:all - //bazel-testlogs/...:all - //bazel-atcoder/...:all - //.ijwb/...:all)" --
```
it fails both on zsh and bash like
```
 ❯ /usr/local/bin/bazel query --tool_tag=ijwb:IDEA:community --output=label_kind --keep_going "attr(\"tags\", \"^((?!manual).)*$\", //...:all - //bazel-bin/...:all - //bazel-genfiles/...:all - //bazel-out/...:all - //bazel-testlogs/...:all - //bazel-atcoder/...:all - //.ijwb/...:all)" --
zsh: event not found: manual).)
```

```
bash-3.2$ /usr/local/bin/bazel query --tool_tag=ijwb:IDEA:community --output=label_kind --keep_going "attr(\"tags\", \"^((?!manual).)*$\", //...:all - //bazel-bin/...:all - //bazel-genfiles/...:all - //bazel-out/...:all - //bazel-testlogs/...:all - //bazel-atcoder/...:all - //.ijwb/...:all)" --
bash: !manual: event not found
```

I think the command works well in many cases, but potentially have some failure trouble with around quotes.

### Suggestion
I feel like doublequote of `attr(..........)` should be single quote.

**Comment by fujigon** [comment]
I known WIndows is not officially supported https://github.com/bazelbuild/intellij/issues/113.
I think this problem is occurred only on Windows for now, but the root cause might affect to any environment.

**Comment by khogeland** [comment]
@fujigon I'm having the same issue (broken with `derive_targets_from_directories`, no error reported). Your branch didn't work for me. What is `print-debug`? How did you find the error?

**Comment by khogeland** [comment]
I found the description here a little confusing and was having the same problem, so I poked around.

For clarification, zsh/bash are not involved- the plugin uses ProcessBuilder which uses the OS' create process API, it does not use the system shell. The sync console output is also misleading. It uses `ParametersListUtil` to join the command list, which does its own quoting/escaping, so the command that gets printed out is not exactly the one that gets passed to the OS.

The issue is that Windows handles escape sequences and quotes in process arguments differently than Mac/Linux. The correct platform-agnostic way to fix this is to use single quotes as fujigon proposed. This is a super simple fix so I hope it can be pulled in soon.
